easybuild/tools/module_generator.py

```diff
@@ -426,24 +434,25 @@ def unpack_setenv_value(self, env_var_name, env_var_val):
         """
         Unpack value that specifies how to define an environment variable with specified name.
         """
-        use_pushenv = False
+        use_pushenv = self.DEFAULT_MODEXTRAVARS_PUSHENV
+        shell_vars = self.DEFAULT_MODEXTRAVARS_SHELL_VARS
 
         # value may be specified as a string, or as a dict for special cases
         if isinstance(env_var_val, str):
             value = env_var_val
-
         elif isinstance(env_var_val, dict):
-            use_pushenv = env_var_val.get('pushenv', False)
+            use_pushenv = env_var_val.get('pushenv', self.DEFAULT_MODEXTRAVARS_PUSHENV)
+            shell_vars = env_var_val.get('shell_vars', self.DEFAULT_MODEXTRAVARS_SHELL_VARS)
             try:
                 value = env_var_val['value']
-            except KeyError:
+            except KeyError as err:
                 raise EasyBuildError("Required key 'value' is missing in dict that specifies how to set $%s: %s",
-                                     env_var_name, env_var_val)
+                                     env_var_name, env_var_val) from err
         else:
             raise EasyBuildError("Incorrect value type for setting $%s environment variable (%s): %s",
                                  env_var_name, type(env_var_val), env_var_val)
 
-        return value, use_pushenv
+        return value, use_pushenv, shell_vars
```

Comment:
Hmm, this changes the function signature of this method, which is a breaking change to the API...

In theory (though I admit it's unlikely) there could be hooks/easyblocks/etc that use the `unpack_setenv_value` method, which would break because of this change.

Reply:
I'll look into making `unpack_setenv_value` an internal method, by prefixing it with `_`, and then making the public method `unpack_setenv_value` deprecated (which allows us to avoid breakage actually)